conformance/lang/expressions/error_constructor.balt

```diff
@@ -0,0 +1,526 @@
+    int x3 = err3; // @error incompatible types: expected 'int', found 'error'
+
+    var err4 = error("error message", error("ERROR!"), msg = "ERROR!", fatal = false);
+    int x4 = err4; // @error incompatible types: expected 'int', found 'error'
+}
+
+Test-Case: error
+Description: Test the first positional-arg is of type string, and the second positional-arg is of type error?
+Labels: error-constructor-expr, module-type-defn, record-type, error-type, optional-type, string
+
+type E error<record {|string message?;|}>;
+
+function errorFunction() {
+    error err = error("error message");
+
+    error err1 = error(err); // @error incompatible types: expected 'string', found 'error'
+    error err2 = error("error message", "error cause"); // @error incompatible types: expected 'error?', found 'string'
+
+    error err3 = error E(err); // @error incompatible types: expected 'string', found 'error'
+    error err4 = error E("error message", "error cause"); // @error incompatible types: expected 'error?', found 'string'
+}
+
+Test-Case: output
+Description: Test the second positional-arg is by default nil and specified type otherwise.
+Labels: error-constructor-expr, module-type-defn, record-type, error-type, exact-equality, error:message, error:cause,
+        nil-literal, string, method-call-expr
+
+type E error<record {|string message?;|}>;
+
+function init() {
+    error err1 = error("error message 1");
+    io:println(err1.message() == "error message 1"); // @output true
+    io:println(err1.cause() == ()); // @output true
+
+    error err = error("error message");
+    error err2 = error("error message 2", err);
+    io:println(err2.message() == "error message 2"); // @output true
+    io:println(err2.cause() == ()); // @output false
+    io:println(err2.cause() === err); // @output true
+
+    error err3 = error E("error message 3");
+    io:println(err3.message() == "error message 3"); // @output true
+    io:println(err3.cause() == ()); // @output true
+
+    error err4 = error E("error message 4", err);
+    io:println(err4.message() == "error message 4"); // @output true
+    io:println(err4.cause() == ()); // @output false
+    io:println(err4.cause() === err); // @output true
+}
+
+Test-Case: output
+Description: Test evaluating the error-constructor-expr constructs a new immutable detail mapping,
+             where each named-arg specifies a field of the error detail mapping and
+             the static type of each named-arg must be a subtype of value:Cloneable.
+Labels: error-constructor-expr, module-type-defn, record-type, error-type, string, is-expr, map-type, intersection-type,
+         readonly, error:detail, value:Cloneable, method-call-expr, list-constructor-expr, DecimalFloatingPointNumber
+
+type E error<record {|string message; boolean fatal?;|}>;
+
+function init() {
+    error err1 = error("error message 1");
+    io:println(err1.detail() is map<value:Cloneable>); // @output true
```

Comment:
Using `value:Cloneable` requires explicitly importing the `lang.value` module. But ATM we don't include the import statements in our tests and it's not added by the test runner too. What should we do in such cases? @pcnfernando

Reply:
Can use `error:Cloneable` once https://github.com/ballerina-platform/ballerina-lang/issues/35182 is fixed.

Reply:
Since this is fixed, shall we update the usages or `value:Cloneable`?